jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/LoadingCacheLayoutModel.java

```diff
@@ -17,47 +17,78 @@
   protected LoadingCache<N, P> locations =
       CacheBuilder.newBuilder().build(CacheLoader.from(() -> pointModel.newPoint(0, 0, 0)));
 
-  public LoadingCacheLayoutModel(LoadingCacheLayoutModel<N, P> other) {
-    super(other.graph, other.pointModel, other.width, other.height, other.depth);
-  }
+  public abstract static class Builder<N, P, T extends LoadingCacheLayoutModel<N, P>> {
+    protected Graph<N> graph;
+    protected PointModel<P> pointModel;
+    protected int width;
+    protected int height;
+    protected int depth;
+
+    protected LoadingCache<N, P> locations =
+        CacheBuilder.newBuilder().build(CacheLoader.from(() -> pointModel.newPoint(0, 0, 0)));
 
-  public LoadingCacheLayoutModel(LoadingCacheLayoutModel<N, P> other, int width, int height) {
-    this(other, width, height, 0);
-  }
+    public LoadingCacheLayoutModel.Builder<N, P, T> setGraph(Graph<N> graph) {
+      this.graph = graph;
+      return this;
+    }
+
+    public LoadingCacheLayoutModel.Builder<N, P, T> setLayoutModel(LayoutModel<N, P> layoutModel) {
+      this.pointModel = layoutModel.getPointModel();
+      this.width = layoutModel.getWidth();
+      this.height = layoutModel.getHeight();
+      this.depth = layoutModel.getDepth();
+      return this;
+    }
+
+    public LoadingCacheLayoutModel.Builder<N, P, T> setPointModel(PointModel<P> pointModel) {
+      this.pointModel = pointModel;
+      return this;
+    }
+
+    public LoadingCacheLayoutModel.Builder<N, P, T> setSize(int width, int height, int depth) {
+      this.width = width;
+      this.height = height;
+      this.depth = depth;
+      return this;
+    }
 
-  public LoadingCacheLayoutModel(
-      LoadingCacheLayoutModel<N, P> other, int width, int height, int depth) {
-    super(other.graph, other.pointModel, width, height, depth);
+    public LoadingCacheLayoutModel.Builder<N, P, T> setSize(int width, int height) {
```

Comment:
I don't agree, since we always set both.

Reply:
The reason I'm recommending this is because that way it's a lot harder to accidentally swap the width and height parameters in the builder.  With this design you have to know which is which.